bndtools.builder/src/org/bndtools/builder/classpath/BndContainerInitializer.java

```diff
@@ -61,15 +64,25 @@
 
     public BndContainerInitializer() {
         super();
-        Central.getInstance().addModelListener(this);
     }
 
     @Override
-    public void initialize(IPath containerPath, IJavaProject javaProject) throws CoreException {
-        IProject project = javaProject.getProject();
+    public void initialize(IPath containerPath, final IJavaProject javaProject) throws CoreException {
+        Central.onWorkspaceInit(new Function<Workspace,Void>() {
+            @Override
+            public Void apply(Workspace a) {
+                try {
+                    Central.getInstance().addModelListener(BndContainerInitializer.this);
```

Comment:
I am not sure if this solves the problem I was raising. Eclipse only creates a single instance of BndContainerInitializer. Not one per projects. So the constructor is only called once and the model listener is only added once. This change calls addModelListener multiple times. But since the method wont add if already added, it is not harmful. However, I think the cleaner solution is to make an onWorkspaceInit call in the constructor to add the model listener rather than piggy backing on the initialize call which is called many times.